src/PoolManager.sol

```diff
@@ -105,6 +105,8 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
         if (key.currency0 >= key.currency1) revert CurrenciesOutOfOrderOrEqual();
         if (!key.hooks.isValidHookAddress(key.fee)) revert Hooks.HookAddressNotValid(address(key.hooks));
 
+        (bool set) = Lockers.setCurrentHook(key.hooks);
```

Comment:
should this be inside the `if` statement?
```
bool set;
if (key.hooks.shouldCallBeforeInitialize()) {
    set = Lockers.setCurrentHook(key.hooks);
    if (key.hooks.beforeInitialize ... {}
}
```

Reply:
ohhh i guess then youd have to do it inside both hooks if statements
just feels weird to set it if we arent even calling for a hook

Reply:
I think we can make this change if we want in the hook callsites refactor